### PR TITLE
Do Not Merge, 

### PR DESCRIPTION
**Pull request by berndhahnebach**
We are in bug fixing for GSoC project FEM elmer and would like to run the Travis tests ...

cheers bernd

### CODE REVIEW

**Comment by berndhahnebach** [comment]
Does someone know why it does not load the modules on linux? 

```
test_pyimport_all_FEM_modules (TestFem.FemTest) ... /usr/local/Mod/Fem/ 
/usr/local/Mod/Fem/PyObjects/ 
Try importing FemConstraint ... 
ERROR
```

it works on my local machine ?!

File: src/Mod/Fem/Gui/DlgSettingsFemCcx.ui [hunk 1]
```diff
@@ -196,28 +196,57 @@
...
+               <font>
+                <weight>75</weight>
+                <bold>true</bold>
+               </font>
+              </property>
+              <property name="accessibleDescription">
+               <string>sdfsdfsdfds</string>
+              </property>
+              <property name="text">
+               <string>Depreciated</string>
```
Comment (luzpaz) [inline src/Mod/Fem/Gui/DlgSettingsFemCcx.ui]: Shouldn't this be `Deprecated` ?

**Comment by drhooves** [comment]
```
======================================================================
ERROR: test_pyimport_all_FEM_modules (TestFem.FemTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/Mod/Fem/TestFem.py", line 381, in test_pyimport_all_FEM_modules
    __import__('{0}'.format(mod))  # to get an error message what was going wrong
  File "/usr/local/Mod/Fem/FemConstraint.py", line 29, in <module>
    from pivy import coin
ImportError: No module named 'pivy'

----------------------------------------------------------------------
```
Could it be that travis doesn't have pivy installed? It's only required by the GUI, not for the app tests. If so we have to guard the import with if App.GuiUp.

```
======================================================================
ERROR: test_static_freq_analysis (TestFem.FemCcxAnalysisTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/Mod/Fem/TestFem.py", line 572, in test_static_freq_analysis
    solver_ccx2_object = ObjectsFem.makeSolverCalculix('CalculiX')
  File "/usr/local/Mod/Fem/ObjectsFem.py", line 347, in makeSolverCalculix
    import FemCalculix.SolverObject
  File "/usr/local/Mod/Fem/FemCalculix/SolverObject.py", line 177
    print pattern
                ^
SyntaxError: Missing parentheses in call to 'print'
```
Looks like a python3 error?!

**Comment by berndhahnebach** [comment]
> Could it be that travis doesn't have pivy installed? It's only required by the GUI, not for the app tests. If so we have to guard the import with if App.GuiUp.


ATM in Fem Gui modules are seperated from App modules, thus I did not need to guide it by a GuiUp test.

Python3 needs print('hello')

**Comment by berndhahnebach** [comment]
fixed al above ...

**Comment by berndhahnebach** [comment]
damn I just guarded the imports, but we would need to guide the whole ViewProviderClass ?! To not have such problems was one off the reason for splitting all modules in GUI and nonGUI part.

I would guard the whole viewprovider for now ?!

bernd

**Comment by berndhahnebach** [comment]
https://github.com/FreeCAD/FreeCAD/pull/945/commits/57c19859fa07637ecd845eed6a2793d1babc6285

I do not really like it but I'm curious if it passes the tests thus I just implemented it.

bernd

**Comment by drhooves** [comment]
No we don't need to guard the whole view provider proxy. The arch workbench does it the same way, can't tell which way is better but splitting generates alot of modules with just 3-4 lines of code.

**Comment by berndhahnebach** [comment]
`No we don't need to guard the whole view provider proxy. The arch workbench does it the same way, can't tell which way is better but splitting generates alot of modules with just 3-4 lines of code.`

IT was my misstake. The import FreeCAD was missing. It is all in here: See https://github.com/FreeCAD/FreeCAD/pull/945/commits/63d38cb71bfde03f393ac6fb76202758b3498476

ATM py3 seams to fail here: https://github.com/berndhahnebach/FreeCAD_bhb/blob/d01289fa75383f1794d388a6563f4792c2e1f057/src/Mod/Fem/ObjectsFem.py#L348

do not know why?

**Comment by drhooves** [comment]
It's because in python3 print is a function, is it not? We could put from __future__ import print_function at the beginning of the file. That would make python2 support the new syntax too.

**Comment by berndhahnebach** [comment]
sure, if there are prints without brackets py3   fails. But AFAIK I fixed all prints. Is there still an old print without brackets somewhere in your code? 

bernd

**Comment by berndhahnebach** [comment]
passes py2 and py3 on linux :-) https://travis-ci.org/FreeCAD/FreeCAD/builds/273963271?utm_source=github_status&utm_medium=notification

**Comment by berndhahnebach** [comment]
:-)

**Comment by sasobadovinac** [comment]
👍 is it ready to merge?

File: src/Mod/Fem/ObjectsFem.py [hunk 2]
```diff
@@ -303,7 +303,7 @@ def makeMeshNetgen(doc, name="FEMMeshNetgen"):
 
 
 def makeMeshRegion(doc, base_mesh, element_length=0.0, name="FEMMeshRegion"):
-    '''makeMeshRegion(document, base_mesh, [element_length] [name]): creates a  FEM mesh region object to define properties for a regon of a FEM mesh'''
+    '''makeMeshRegion(document, base_mesh, [element_length], [name]): creates a  FEM mesh region object to define properties for a regon of a FEM mesh'''
```
Comment (luzpaz) [inline src/Mod/Fem/ObjectsFem.py]: @berndhahnebach regon->region

Comment (berndhahnebach) [reply src/Mod/Fem/ObjectsFem.py]: https://github.com/FreeCAD/FreeCAD/pull/945/commits/4e40c946

**Comment by berndhahnebach** [comment]
continues here:  https://github.com/FreeCAD/FreeCAD/pull/1076

due to commit https://github.com/FreeCAD/FreeCAD/commit/ce68094 a rebase makes no sence anymore ...

**Pull request closed without merging**